platform/qt/src/qmapboxgl.cpp

```diff
@@ -964,15 +1152,30 @@ void QMapboxGL::updateSource(const QString &sourceID, const QVariantMap &params)
     }
 }
 
-void QMapboxGL::removeSource(const QString& sourceID)
+/*!
+    Removes the source \a id.
+
+    Has no effect if the source does not exist.
+*/
+void QMapboxGL::removeSource(const QString& id)
 {
-    auto sourceIDStdString = sourceID.toStdString();
+    auto sourceIDStdString = id.toStdString();
 
     if (d_ptr->mapObj->getSource(sourceIDStdString)) {
         d_ptr->mapObj->removeSource(sourceIDStdString);
     }
 }
 
+/*!
+    Adds a custom layer \a id with the initialization function \a initFn, the
+    render function \a renderFn and the deinitialization function \a deinitFn with
+    the user data \a context_ before the existing layer \a before.
+
+    This is used for delegate the rendering to the user of this API and is not
+    officially supported. Use at your own risk.
+
+    \deprecated
```

Comment:
Deprecating this method implies that it could be removed in a future release, not that it could change at any time. Changing or removing a deprecated method is usually regarded as a backwards-incompatible change to the library. (In the case of semver, that means a major revision bump.)

In the iOS and macOS SDKs, we’ve signaled that the MGLOpenGLStyleLayer API is not officially supported by leaving it undocumented (at least undocumented in public headers). This is consistent with a longstanding Apple convention, but I wonder if there’s a similar convention in Qt-land that calling an undocumented method is discouraged.